File: backend-go/src/models/item.go
package models

import "go.mongodb.org/mongo-driver/bson/primitive"

type Item struct {
	ID               primitive.ObjectID `bson:"_id,omitempty" json:"_id,omitempty"`
	Name             string             `bson:"name" json:"name"`
	Barcode          string             `bson:"barcode" json:"barcode"`
	IsValidBarcode   bool               `bson:"isValidBarcode" json:"isValidBarcode"`
	IsBarcodeChecked bool               `bson:"isBarcodeChecked" json:"isBarcodeChecked"`
	Price            float64            `bson:"price" json:"price"`
	Stock            float64            `bson:"stock" json:"stock"`
	User             primitive.ObjectID `bson:"user" json:"user"`
	OffPrice         float64            `bson:"offPrice" json:"offPrice"`
	ExpiryDates      []int64            `bson:"expiryDates" json:"expiryDates"`
}

File: backend/models/Item.js
const mongoose = require('mongoose');

const itemSchema = new mongoose.Schema({
	name: {
		type: String,
		required: [true, 'Name is required']
	},
	barcode: {
		type: String,
		default: ""
	},
	isValidBarcode: {
		type: Boolean,
		default: false
	},
	isBarcodeChecked: {
		type: Boolean,
		default: false
	},
	price: {
		type: Number,
		required: [true, 'Price is required']
	},
	stock: {
		type: Number,
		required: [true, 'Stock is required']
	},
	user: {
		type: mongoose.Schema.Types.ObjectId,
		ref: 'User',
		required: [true, 'A user/ownser is required']
	},
	offPrice: {
		type: Number,
		default: 0
	},
	expiryDates: [Number] // UNIX codes
});

module.exports = mongoose.model('Item', itemSchema);
